networking/p2p/src/main/java/tech/pegasys/teku/networking/p2p/libp2p/MuxFirewall.java

```diff
@@ -115,7 +128,18 @@ public void channelRead(final ChannelHandlerContext ctx, final Object msg) {
         remoteOpenedStreamIds.remove(muxFrame.getId());
       }
       if (!blockFrame) {
-        ctx.fireChannelRead(msg);
+        try {
+          ctx.fireChannelRead(msg);
+        } catch (final Throwable e) {
```

Comment:
All the exceptions inside `ChannelHandler` callbacks are caught and reported to `exceptionCaught()` callback (first the same handler, then if needed downstream - to next handlers towards the wire). So you wouldn't catch the exception here. 

We likely need to sort out and refactor handling of exceptions for inbound traffic in libp2p.